frontend/pages/policies/ManagePoliciesPage/components/InstallSoftwareModal/InstallSoftwareModal.tsx

```diff
@@ -152,19 +158,62 @@ const InstallSoftwareModal = ({
     [formData]
   );
 
-  const availableSoftwareOptions = titlesAFI?.map((title) => {
-    const splitName = title.software_package?.name.split(".") ?? "";
-    const ext =
-      splitName.length > 1 ? splitName[splitName.length - 1] : undefined;
-    const platformString = ext
-      ? `${getPlatformDisplayFromPackageExtension(ext)} (.${ext}) • `
-      : "";
-    return {
-      label: title.name,
-      value: title.id,
-      helpText: `${platformString}${title.software_package?.version ?? ""}`,
-    };
-  });
+  /**
+   * Filters and transforms software titles into dropdown options
+   * Filters titlesAFI to include only software compatible with the policy's
+   * platform(s) by using the file extension to determine the platform
+   */
+  const availableSoftwareOptions = (policy: IFormPolicy) => {
+    return titlesAFI
+      ?.filter((title) => {
+        const splitName = title.software_package?.name.split(".") ?? "";
+        const ext =
+          splitName.length > 1 ? splitName[splitName.length - 1] : undefined;
+        const platform = getPlatformFromExtension(ext);
```

Comment:
Thinking we can unify/simplify the filtering by having:

1. Platform list (splitting commas) defined ~L167.
2. `packagePlatform = PLATFORMS_BY_SOURCE[title?.source] ?? null` which works for both VPP and installer-based titles
3. result of step 1 includes result of step 2 for the filter

Reply:
We don't have a `PLATFORM_BY_SOURCE` conversion. I tried to update our interface for source since I realized it was missing linux sources, but theres a ton... I think they're all the keys of `SOURCE_TYPE_CONVERSION` but I don't have a map to platform, so I'm just going to keep using the goofy `getPlatformFromExtension` and leave a note.

Reply:
```
export const SOURCE_PLATFORM_CONVERSION = {
  apt_sources: "linux",
  deb_packages: "linux",
  portage_packages: "linux",
  rpm_packages: "linux",
  yum_sources: "linux",
  apps: "darwin",
  ios_apps: "ios",
  ipados_apps: "ipados",
  homebrew_packages: "darwin",
  programs: "windows",
  chocolatey_packages: "windows",
  pkg_packages: "macos",
} as const;
```

Shorter list that's still a superset of software titles with installers that we'll actually see

Reply:
TODO comment committed here: https://github.com/fleetdm/fleet/pull/25202/commits/e823c3ac380e561a647ee58c4706de57483088a5

Reply:
@iansltx goat. I'll put it in Monday

Reply:
@iansltx I am working in the sky today about to commit this update. I just realized if we are improving this to find the platform from the source, we need to update this tooltip. Ian, could you confirm this change and new tooltip with @eugkuo ?

<img width="991" alt="Screenshot 2024-12-27 at 12 40 41 PM" src="https://github.com/user-attachments/assets/d4851bfe-4e93-4928-a82c-d43e7cf059a9" />